src/components/Dashboard/widgets/Embeds/WalletAddressesEmbed.tsx

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+
+import styled from '@emotion/styled';
+
+import { dispatch } from 'store/rematch';
+import { selectAccountAddress, selectAccountPubKey, selectCurrentTokenInfo } from 'store/selectors';
+import { V } from 'util/theming';
+import { ModalName, ResourceType } from 'vars/defines';
+
+import { Button } from 'components/_General/buttons';
+import CopyToClipboard from 'components/_General/CopyToClipboard';
+import { ReceiveModalOpts } from 'components/Modal/content/Receive';
+import { EmbedContentContainer, RowWrapper } from '../common';
+
+const WalletAddressesEmbedRoot = styled(EmbedContentContainer)`
+  padding: 20px 30px;
+`;
+
+const WalletAddressWidgetLabel = styled.p`
+  text-transform: uppercase;
+  font-size: ${V.font.pSmall};
+  color: ${V.color.frontSoft};
+`;
+
+const Copy = styled.span`
+  padding-top: 11px;
+  margin-left: 5px;
+`;
+
+type WalletAddressWidgetProps = {
+  title: string;
+  modal_type: string;
+};
+
+const openWalletModal = (options: ReceiveModalOpts) => () =>
+  dispatch.environment.SET_MODAL({ name: ModalName.RECEIVE, options });
+
+const DisplayWalletAddress = ({ title, modal_type }: WalletAddressWidgetProps) => {
+  const tokenInfo = useSelector(selectCurrentTokenInfo);
+  const isNFT = tokenInfo && tokenInfo.supply === 1;
```

Comment:
feel free to use optional chaining here
```suggestion
  const isNFT = tokenInfo?.supply === 1;
```